cluster/examples/coreos/ceph-reboot-check-script.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ceph-reboot-check-script
+  namespace: rook
+data:
+  status-check.sh: |-
+    #!/bin/bash
+    # Wait 30 second to be sure ceph status has latest info
+    echo "ROOK-UPDATE-SCRIPT:: Running the rook initiation script..."
+    /usr/local/bin/toolbox.sh &
+
+    echo "ROOK-UPDATE-SCRIPT:: Sleeping for 30 seconds to make sure Ceph has the latest cluster status (see ROOK_MON_HEALTHCHECK_INTERVAL in the rook-operator manifest)"
+    /bin/sleep 30
+
+    # Following command looks in the ceph status command for the message 'status: HEALTH_OK'
+
+    echo "ROOK-UPDATE-SCRIPT:: Running the Ceph status command"
+    /usr/bin/ceph status -f json | grep '"status":"HEALTH_OK"' &> /dev/null
+    if [ $? == 0 ]; then
+
+       echo "ROOK-UPDATE-SCRIPT:: Cluster Healthy, we can post to Kube API"
+
+       export CODE="[{ \"op\": \"add\", \"path\": \"/metadata/annotations/ceph-reboot-check\", \"value\": \"true\" }]"
+       export KUBE_TOKEN=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
+       curl -sSk -H "Authorization: Bearer $KUBE_TOKEN" -XPATCH -H "Accept: application/json" -H "Content-Type:application/json-patch+json"  --data "$CODE" https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/api/v1/nodes/$NODE_NAME
+
+       echo "ROOK-UPDATE-SCRIPT:: T$NODE_NAME annotations are updated, reboot permissioned!"
+    else
+       echo "ROOK-UPDATE-SCRIPT:: The Ceph cluster is not healthy, this pod will repeat until the status is healthy"
+       exit 200
```

Comment:
The script should not just exit. In the end you have a `CrashloopBackOff` Pod which will take longer and longer to restart.
Making the script into a loop and only exiting on failures would be good.
If you have questions about that let me know.

Reply:
Do you suggest that on success the script enters an infinite loop?

Reply:
Yes, as written when it exits the Pod will enter a CrashLoopBackOff at some point after too many Pod "restarts".